In [2]:
from pymongo import MongoClient
from pymongo.errors import ConnectionFailure

##Create URI 
MONGO_URI= f'mongodb+srv://Lejones:tAUBA8pNu1Eq4bKe@finalproject.ngck3vj.mongodb.net/'

#Constants
DB_NAME = 'FinalDB'
COLLECTION_NAME1 = 'Walkability_Collection_fixed'
COLLECTION_NAME2 = 'AQ_Collection'

#initializing 
client = MongoClient(MONGO_URI)
db = client[DB_NAME]
walk = db[COLLECTION_NAME1]
AQ= db[COLLECTION_NAME2]



In [23]:
# Making sure collections are accessible 
results = walk.find().limit(5)
for r in results:
    print(r)

results = AQ.find().limit(5)
for r in results:
    print(r)

{'_id': ObjectId('6940727adcc2fcea5096f748'), 'STATEFP': '48', 'COUNTYFP': '113', 'TRACTCE': '7825', 'BLKGRPCE': '4', 'STC_Fips': '48113', 'State_Names': 'Texas', 'Total_area_AC': 73.5950285, 'Total_water_AC': 0.0, 'Total_land_AC': 73.5950285, 'Total_unprotected_land_AC': 73.5950285, 'Total_pro_land_AC': 0.0, 'TotPop': 1202, 'Housing_Units': 460.0, 'Occupied_HU': 423.0, 'WorkingAge_PCT': 0.549, '0_CarsOwned': 69, '0Cars_PCT': 0.163120567, '1_CarsOwned': 39, '1Cars_PCT': 0.092198582, '2more_CarsOwned': 315, '2pCars_PCT': 0.744680851, 'Workers': 412, 'CountLowWage': 99, 'CountMedWage': 122, 'CountHiWage': 191, 'LowWage_PCT': 0.240291262, 'MedWage_PCT': 0.2961165048543689, 'HighWage_PCT': 0.46359223300970875, 'TotEmp': 66, 'Emp_CountLowWage': 21, 'Emp_CountMedWage': 27, 'Emp_CountHiWage': 18, 'Emp_LowWage_PCT': 0.318181818, 'Total_Activity': -394.0, 'Housing_Density': 6.250422201, 'Pop_Density': 16.33262497, 'Emp_Density': 0.896799707, 'PopandEmp_Density': 7.147221908, 'Emp_per_Household'

# Which cities or counties rank as the most walkable in the country, and what factors contribute to their scores? 

In [3]:

pipeline = [
    {"$sort": {"NatWalkInd":-1}}, # higher walkability is better 
    {'$limit':10},
    {"$project": {"_id": 0}}
]
res = walk.aggregate(pipeline)

for r in res:
    print(r)

{'STATEFP': '4', 'COUNTYFP': '19', 'TRACTCE': '400', 'BLKGRPCE': '3', 'STC_Fips': '419', 'State_Names': 'Arizona', 'Total_area_AC': 206.8973242, 'Total_water_AC': 0.0, 'Total_land_AC': 206.8973242, 'Total_unprotected_land_AC': 206.89276, 'Total_pro_land_AC': 0.004564200000004348, 'TotPop': 1171, 'Housing_Units': 684.0, 'Occupied_HU': 583.0, 'WorkingAge_PCT': 0.738, '0_CarsOwned': 91, '0Cars_PCT': 0.156089194, '1_CarsOwned': 329, '1Cars_PCT': 0.56432247, '2more_CarsOwned': 163, '2pCars_PCT': 0.279588336, 'Workers': 434, 'CountLowWage': 124, 'CountMedWage': 172, 'CountHiWage': 138, 'LowWage_PCT': 0.285714286, 'MedWage_PCT': 0.39631336405529954, 'HighWage_PCT': 0.31797235023041476, 'TotEmp': 1652, 'Emp_CountLowWage': 576, 'Emp_CountMedWage': 764, 'Emp_CountHiWage': 312, 'Emp_LowWage_PCT': 0.348668281, 'Total_Activity': 968.0, 'Housing_Density': 3.306060589, 'Pop_Density': 5.659937061, 'Emp_Density': 7.984813002, 'PopandEmp_Density': 11.29087359, 'Emp_per_Household': 2.833619211, 'Emp_Hous

In [12]:
# Group by walkability and look into protected land, Income percentages
pipeline= [
    {"$match":{"NatWalkInd":{"$gte": 18}}},
    {"$sort": {"NatWalkInd": 1}},
    {"$project": {"_id":0, "NatWalkInd":1, "Total_pro_land_AC":1}}
 ]
res = walk.aggregate(pipeline)

for r in res:
    print(r)

{'Total_pro_land_AC': 7.384105300000016, 'NatWalkInd': 18.0}
{'Total_pro_land_AC': 0.0, 'NatWalkInd': 18.0}
{'Total_pro_land_AC': 0.0, 'NatWalkInd': 18.0}
{'Total_pro_land_AC': 0.0, 'NatWalkInd': 18.0}
{'Total_pro_land_AC': 0.0, 'NatWalkInd': 18.0}
{'Total_pro_land_AC': 10.883425599999995, 'NatWalkInd': 18.0}
{'Total_pro_land_AC': 0.0, 'NatWalkInd': 18.0}
{'Total_pro_land_AC': 3.4047313000000017, 'NatWalkInd': 18.0}
{'Total_pro_land_AC': 26.557647400000008, 'NatWalkInd': 18.0}
{'Total_pro_land_AC': 2.3771495999999956, 'NatWalkInd': 18.0}
{'Total_pro_land_AC': 0.0, 'NatWalkInd': 18.0}
{'Total_pro_land_AC': 0.0, 'NatWalkInd': 18.0}
{'Total_pro_land_AC': 0.5604708999999986, 'NatWalkInd': 18.0}
{'Total_pro_land_AC': 0.0, 'NatWalkInd': 18.0}
{'Total_pro_land_AC': 29.45466349999998, 'NatWalkInd': 18.0}
{'Total_pro_land_AC': 0.0, 'NatWalkInd': 18.0}
{'Total_pro_land_AC': 219.28557139999998, 'NatWalkInd': 18.0}
{'Total_pro_land_AC': 9.014280299999982, 'NatWalkInd': 18.0}
{'Total_pro_land_AC': 

In [ ]:
# find communities with greater high wages and see their walkability 

pipeline = [
    {"$match":{"$expr":{"$gt":["$HighWage_PCT", "$LowWage_PCT"]}}},
    {"$count": "HighWage Earning Communities"}
]

res = walk.aggregate(pipeline)

for r in res:
    print(r)


{'HighWage Earning Communities': 184905}


In [ ]:
#find average walkability of higher wage communities 

match = {"$match":{"$expr":{"$gt":["$HighWage_PCT", "$LowWage_PCT"]}}}
group={"$group": {"_id": "null", "Average_Walkability": {"$avg":"$NatWalkInd"}}}


res = walk.aggregate([match, group])

for r in res:
    print(r)

{'_id': 'null', 'Average_Walkability': 9.512679484068245}


In [ ]:
#find average walkability of lower wage communities
match = {"$match":{"$expr":{"$gt":["$LowWage_PCT", "$HighWage_PCT"]}}}
group={"$group": {"_id": "null", "Average_Walkability": {"$avg":"$NatWalkInd"}}}


res = walk.aggregate([match, group])

for r in res:
    print(r)

{'_id': 'null', 'Average_Walkability': 10.394180993155334}


To answer this question I first found where the highest areas of walkability are. Within the top 10, 6 are in California. This coincides with our graph of the Average High Wage and Low Wage workers depending on Walkability. The visualization shows that the more walkable a place is the higher percentage of High wage workers live there and California, with having the largest economy in the united states, follows this trend. These both suggest a correlation between the wages of the people who live there and the Walkability. I go onto explore this idea but first I check and see if the total protected land of an area changes the Walkability, which it does not seem to do. I investigated this by checking the Total protected land of areas with high walkability and upon reviewing the values there seemed to be no correlation. I then went on to investigate the relationship between the wages and Walkability. I did this by finding the average walkability of the higher wage communites and Lower wage communites and determined which were which by using a match stage. For high wage communites the match selected entries that had a greater percentage of high wage workers than low wage workers and for lower wage it was the reverse. Upon finding the average of these two groups you can see that they are quite similar. There is slightly higher average Walkability within low wage communities but this is contradicted and given more context by our Visualization. 

This Visualization shows us that no matter the Walkability there isn't much fluctuation between the amount of Low wage workers in a region however there is a positive correlation between the number of high wage workers in a region and higher Walkability. This may suggest two things, one that people who have a higher income and can afford to move, choose to move to areas with greater walkability, or that economically prosperous areas that have high wage jobs and need more High wage workers, like cites, are in turn more walkable. 

# How does public transport access affect Walkability 

In [ ]:
#find the top public transit access and the lowest and compare their walkability scores
#best transit access
pipeline= [
    {"$sort": {"Ranked_TranDist": -1}},
    {"$limit": 20},
    {"$group":{ "_id" : "null" , "AVG Walkability": {"$avg": "$NatWalkInd"}}}
]

res = walk.aggregate(pipeline)
for r in res:
    print(r)


{'_id': 'null', 'AVG Walkability': 14.833333333999999}


In [ ]:
#worst transit access 
pipeline= [
    {"$sort": {"Ranked_TranDist": 1}},
    {"$limit": 20},
    {"$group":{ "_id" : "null" , "AVG Walkability": {"$avg": "$NatWalkInd"}}}
]
res = walk.aggregate(pipeline)
for r in res:
    print(r)

{'_id': 'null', 'AVG Walkability': 8.38333333345}


To answer this question calculated the average walkability of the 20 areas with the best transit access and 20 areas with the worst transit access. This revealed a drastic change, showing that areas with poor transit access have much worse Walkability, coming in with and average of 8.4 and areas with good public transit access have considerably better Walkability, coming in at 14.8. 

# How does public transport access affect the amount of cars owned per household

In [ ]:
#best transit access
pipeline= [
    {"$sort": {"Ranked_TranDist": -1}},
    {"$limit": 20},
    {"$group":{ "_id" : "null" , 
               "Total_without_cars": {"$avg": "$0_CarsOwned"},
               "Total_with_1car":{"$avg": "$1_CarsOwned"},
               "Total_with_2cars":{"$avg": '$2more_CarsOwned'}}}
]
res = walk.aggregate(pipeline)
for r in res:
    print(r)

{'_id': 'null', 'Total_without_cars': 68.05, 'Total_with_1car': 328.15, 'Total_with_2cars': 229.6}


In [91]:
#worst transit access
pipeline= [
    {"$sort": {"Ranked_TranDist": 1}},
    {"$limit": 20},
    {"$group":{ "_id" : "null" , 
               "Total_without_cars": {"$avg": "$0_CarsOwned"},
               "Total_with_1car":{"$avg": "$1_CarsOwned"},
               "Total_with_2cars":{"$avg": '$2more_CarsOwned'}}}
]
res = walk.aggregate(pipeline)
for r in res:
    print(r)

{'_id': 'null', 'Total_without_cars': 19.5, 'Total_with_1car': 180.0, 'Total_with_2cars': 455.95}


For this question I once again found the best and worst 20 areas for transit and then found the average amount of people who owned no cars, one car or two or more cars. We can see through the results that the average amount of people without cars increases in areas with good transit access, and in areas with poor transit access the average amount of people with 2 or more cars also greatly increases. 

# How does road network density affect walkability?

In [78]:
#Highest Road density
pipeline= [
    {"$sort": {'Tot_Road_network_Den': -1}},
    {"$limit": 20},
    {"$group":{ "_id" : "null" , 
              "AVG Walkability": {"$avg": "$NatWalkInd"}}}
]
res = walk.aggregate(pipeline)
for r in res:
    print(r)

{'_id': 'null', 'AVG Walkability': 15.5166666665}


In [ ]:
#Lower road Density 
pipeline= [
    {"$sort": {'Tot_Road_network_Den': 1}},
    {"$limit": 20},
    {"$group":{ "_id" : "null" , 
              "AVG Walkability": {"$avg": "$NatWalkInd"}}}
]
res = walk.aggregate(pipeline)
for r in res:
    print(r)

{'_id': 'null', 'AVG Walkability': 4.3833333332999995}


For this question I found the most dense and least dense road systems and calculated the average Walkability. With these results, we can see drastic drop of nearly 11 degrees of walkability between the most dense and least dense areas. This indicates that the road density of an area is a significant factor in the overall walkabilty of an area, even more so than public transit. 

# Average Walkability of Each County 

In [ ]:
pipeline = [
    {"$group": { "_id": "$STC_Fips","AVG Walkability": {"$avg": "$NatWalkInd"} }}
]
res = walk.aggregate(pipeline)

for r in res:
    print(r)

    

{'_id': '1761', 'AVG Walkability': 5.235294117529412}
{'_id': '191', 'AVG Walkability': 5.484848484696969}
{'_id': '1729', 'AVG Walkability': 6.931372548941177}
{'_id': '4955', 'AVG Walkability': 6.583333333500001}
{'_id': '2317', 'AVG Walkability': 5.913194444541666}
{'_id': '42101', 'AVG Walkability': 14.847264056201807}
{'_id': '21129', 'AVG Walkability': 4.666666666714286}
{'_id': '89', 'AVG Walkability': 5.66666666675}
{'_id': '18147', 'AVG Walkability': 5.55208333325}
{'_id': '3427', 'AVG Walkability': 9.987005649772883}
{'_id': '21181', 'AVG Walkability': 4.333333333333333}
{'_id': '1663', 'AVG Walkability': 4.91666666675}
{'_id': '28159', 'AVG Walkability': 4.561403508947368}
{'_id': '3683', 'AVG Walkability': 10.121333333344001}
{'_id': '183', 'AVG Walkability': 9.387286324785256}
{'_id': '5325', 'AVG Walkability': 6.815217391478262}
{'_id': '48471', 'AVG Walkability': 5.526315789473684}
{'_id': '37139', 'AVG Walkability': 8.17999999968}
{'_id': '55109', 'AVG Walkability': 6.4